jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/exception/QueryExceptionHandlerTest.java

```diff
@@ -15,65 +15,140 @@
  */
 package com.salesforce.datacloud.jdbc.exception;
 
+import static com.salesforce.datacloud.jdbc.hyper.HyperTestBase.getHyperQueryConnection;
+import static org.junit.Assert.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertInstanceOf;
 
+import com.salesforce.datacloud.jdbc.core.DataCloudStatement;
+import com.salesforce.datacloud.jdbc.hyper.HyperTestBase;
 import com.salesforce.datacloud.jdbc.util.GrpcUtils;
 import io.grpc.StatusRuntimeException;
 import java.sql.SQLException;
+import java.util.Properties;
 import lombok.val;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
+@ExtendWith(HyperTestBase.class)
 class QueryExceptionHandlerTest {
-
     @Test
-    public void testCreateExceptionWithStatusRuntimeException() {
-        StatusRuntimeException fakeException = GrpcUtils.getFakeStatusRuntimeExceptionAsInvalidArgument();
-        SQLException actualException = QueryExceptionHandler.createException("test message", fakeException);
-
-        assertInstanceOf(SQLException.class, actualException);
-        assertEquals("42P01", actualException.getSQLState());
-        val sep = System.lineSeparator();
-        assertEquals(
-                "42P01: Table not found" + sep + "DETAIL:" + sep + sep + "HINT:" + sep, actualException.getMessage());
-        assertEquals(StatusRuntimeException.class, actualException.getCause().getClass());
+    public void testServerReportedErrorWithCustomerDetails() throws SQLException {
+        // Verify that the normal exception messages contains the query and customer details & hints from the server
+        // error info
+        try (val connection = getHyperQueryConnection()) {
+            try (val stmt = (DataCloudStatement) connection.createStatement()) {
+                String query = "WITH \"A\" AS (SELECT 1) SELECT * FROM A";
+                DataCloudJDBCException ex = assertThrows(DataCloudJDBCException.class, () -> stmt.executeQuery(query));
+                String customerDetail = "\n"
+                        // Indentation to more easily see that the ascii art matches (the mismatch is from the escape
+                        // characters for the double quotes)
+                        + "line 1, column 38: WITH \"A\" AS (SELECT 1) SELECT * FROM A\n"
+                        + "                                                        ^";
+                String customerHint = "Try quoting the identifier: `\"A\"`";
+                String expectedMessage = "Failed to execute query: table \"a\" does not exist\n" + "SQLSTATE: 42P01\n"
+                        + "QUERY-ID: "
+                        + stmt.getQueryId() + "\n" + "DETAIL: "
+                        + customerDetail
+                        + "\n" + "HINT: "
+                        + customerHint;
```

Comment:
the placement of the `"\n"` newlines is pretty arbitrary...
Can we use string interpolation/string templates + text blocks?